### PR TITLE
Factor asset filePaths when computing bundle hashes

### PR DESCRIPTION
**Pull request by wbinnssmith**
Currently it's possible for bundles to produce identical hashes if the output contents of their assets are identical. Factor in asset filePaths as well.

In the future we may be able to deduplicate bundles entirely based on their packaged and optimized content as well as source map content.

Is there anything else we should include in the bundle hash? #3545 also includes the bundle's environment.

Test Plan: Added an integration test. Verified it failed before applying the patch.

### CODE REVIEW

**Comment by garthenweb** [comment]
We had a bug in parcel v1 where a dependency was updated in the main bundle which was also referenced in a child bundle, so the id changed in both bundles but the hash of the child bundle was not updated as the real content did not change. As we cached those JS files some users had a "module not found" error.

I am not sure if this is relevant for v2 (especially with tree shaking) and it also didn't happen always (didn't found out the exact cause so far). But maybe it is worth to add the IDs of the dependencies into the hash?

I can check on the weekend if this is still an issue with the latest v2 version (this commit) and create a bug report/ example project if needed!

**Comment by garthenweb** [comment]
BTW: I think this PR might fix https://github.com/parcel-bundler/parcel/issues/3442 .

File: packages/core/core/src/BundleGraph.js [hunk 1]
```diff
@@ -518,7 +518,7 @@ export default class BundleGraph {
     let hash = crypto.createHash('md5');
     // TODO: sort??
     this.traverseAssets(bundle, asset => {
-      hash.update(asset.outputHash);
+      hash.update([asset.outputHash, asset.filePath].join(':'));
```
Comment (DeMoorJasper) [inline packages/core/core/src/BundleGraph.js]: Won't this break sharing cache between machines as this is an absolute path?

Not sure if this is something we wanna support.

Comment (mischnic) [reply packages/core/core/src/BundleGraph.js]: > Not sure if this is something we wanna support.

Will was working on a HTTP remote cache backend, so I would say yes.

**Comment by garthenweb** [comment]
I didn't manage to set up an example which bundles node modules with the v2 branch, but here is an example using parcel v1 which illustrates the issue mentioned above: https://github.com/garthenweb/parcel-bundle-names-example Maybe it is useful, let me know if there are questions :)

File: packages/core/core/src/BundleGraph.js (hunk 1 above)
Comment (wbinnssmith) [reply packages/core/core/src/BundleGraph.js]: Yeah, we'll definitely need filepaths on the assets themselves to be relative to something like the project root.

**Pull request merged**